include/library

```diff
@@ -0,0 +1,247 @@
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# shellquote STRING
+#
+# Quote STRING so that STRING == $(echo "STRING")
+#
+lib_shellquote () {
+  arg=${1//\\/\\\\}
+  arg=${arg//\$/\\\$}
+  arg=${arg//\"/\\\"}
+  arg=${arg//\`/\\\`}
+  echo -n "$arg"
+}
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# lib_set_config FILE KEY VALUE
+#
+# Set config valiable KEY in FILE to VALUE.
+#
+# Note that KEY must already exist.
+#
+lib_set_config ()
+{
+  err=0
+
+  if [ -w "$1" ] ; then
+    sed -i -E -e "s/^($2=)\S+/\1\"$3\"/" "$1"
+    err=$?
+  else
+    echo "$1: not writable" >&2
+    err=1
+  fi
+
+  return "$err"
+}
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# lib_get_cmdline_grub
+#
+# Set $cmdline to default grub cmdline.
+#
+lib_get_cmdline_grub ()
+{
+  if [ -e /etc/default/grub ] ; then
+    . /etc/default/grub
+    if [ -z "$GRUB_CMDLINE_LINUX_DEFAULT" ] ; then
+      echo "Warning: GRUB_CMDLINE_LINUX_DEFAULT is empty"
+    fi
+  else
+    echo "/etc/default/grub missing"
+    exit 1
+  fi
+
+  cmdline="$GRUB_CMDLINE_LINUX_DEFAULT"
+}
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# lib_set_cmdline_grub
+#
+# Set default grub cmdline to $cmdline.
+#
+lib_set_cmdline_grub ()
+{
+  tmp_cmdline=${cmdline//\\/\\\\\\\\}
+  tmp_cmdline=${tmp_cmdline//\"/\\\\\"}
+
+  sed -i -e "/^GRUB_CMDLINE_LINUX_DEFAULT=/c GRUB_CMDLINE_LINUX_DEFAULT=\"$tmp_cmdline\"" /etc/default/grub || {
+    echo "/etc/default/grub: failed to update config file"
+    exit 1
+  }
+}
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# lib_get_cmdline_bls
+#
+# Set $cmdline to default bls cmdline.
+#
+lib_get_cmdline_bls ()
+{
+  if [ -n "$KERNEL_INSTALL_CONF_ROOT" ] ; then
+    cmdline_file="$KERNEL_INSTALL_CONF_ROOT/cmdline"
+  elif [ -f /etc/kernel/cmdline ] ; then
+    cmdline_file=/etc/kernel/cmdline
+  elif [ -f /usr/lib/kernel/cmdline ] ; then
+    cmdline_file=/usr/lib/kernel/cmdline
+  else
+    cmdline_file=/proc/cmdline
+  fi
+
+  if [ -f "$cmdline_file" ] ; then
+    cmdline=$(cat $cmdline_file)
+
+    lib_del_option BOOT_IMAGE
+    lib_del_option initrd
+    lib_del_option systemd.machine_id
```

Comment:
Removing this will make "lib_get_cmdline_bls" followed by "lib_set_cmdline_bls" break the cmdline.

For example, some images are using machine_id and other not, removing it during the get is a risk to lost it later?